sound/soc/sof/core.c

```diff
@@ -353,6 +375,9 @@ int snd_sof_device_remove(struct device *dev)
 		snd_sof_free_trace(sdev);
 	}
 
+	/* unregister virtbus devices */
+	virtbus_dev_unregister(&sdev->sof_virtbus_dev->vdev);
```

Comment:
What if `sof_probe_continue()` ran on a work queue and failed before registering the virtual device, wouldn't this Oops?

Reply:
@lyakh what would be your suggestion in this case? Should I check if the virtual dev is not NULL before unregistering? Or should virtbus_dev_unregister() be able to handle NULL pointers?

Reply:
@ranj063 you cannot check in `virtbus_dev_unregister()` and its argument as used above won't be NULL - it's an address of an object. But `sdev->sof_virtbus_dev` would be NULL in case of that probing failure, that I mention, so `&(NULL->member)` would resolve to some small positive number, which then will Oops. So you shouldn't be calling this function if `sdev->sof_virtbus_dev` is NULL. You have to follow the common pattern and use the `probe()` / `release()` scheme, used in SOF, i.e. move that call two lines up under the `if (sdev->fw_state > SOF_FW_BOOT_NOT_STARTED)` branch.